core/src/main/java/mod/chiselsandbits/client/model/meshing/GreedyMeshBuilder.java

```diff
@@ -26,172 +25,192 @@ public interface MaterialProvider {
-
-        for (int dimension = 0; dimension < DIMENSIONS; dimension++) {
-            int currentFirstDimensionIter, currentSecondDimensionIter;  // The current iteration position in the x, y, and z dimensions, also known as i, j, and k.
-            int width, height; //The current iteration dimensions in the x, y, and z dimensions, also known as l, w, and h.
-
-            //We need to iterate over the other two dimensions.
-            final int firstDimensionIter = (dimension + 1) % DIMENSIONS; // The u and v dimensions are used to iterate over the other two dimensions.
-            final int secondDimensionIter = (dimension + 2) % DIMENSIONS;
-
-            final int[] dimensionalIterator = new int[]{0, 0, 0};
-            final int[] calculatedDimensionsOffset = new int[]{0, 0, 0}; //Dimension offset for the already calculated dimensions.
-
-            calculatedDimensionsOffset[dimension] = 1;
-            for (dimensionalIterator[dimension] = -1; dimensionalIterator[dimension] < sizePerDimension; ) {
-
-                //Compute mask
-                int maskIndex = 0;
-                for (dimensionalIterator[secondDimensionIter] = 0; dimensionalIterator[secondDimensionIter] < sizePerDimension; ++dimensionalIterator[secondDimensionIter]) {
-                    for (dimensionalIterator[firstDimensionIter] = 0; dimensionalIterator[firstDimensionIter] < sizePerDimension; ++dimensionalIterator[firstDimensionIter], ++maskIndex) {
-
-                        int currentMaterial = (0 <= dimensionalIterator[dimension] ? materialProcessor.getMaterialIndex(
-                                dimensionalIterator[0],
-                                dimensionalIterator[1],
-                                dimensionalIterator[2]) : 0);
-                        int neighborMaterial = (dimensionalIterator[dimension] < sizePerDimension - 1 ? materialProcessor.getMaterialIndex(
-                                dimensionalIterator[0] + calculatedDimensionsOffset[0],
-                                dimensionalIterator[1] + calculatedDimensionsOffset[1],
-                                dimensionalIterator[2] + calculatedDimensionsOffset[2]) : 0);
-
-                        boolean aIsTruthy = currentMaterial != 0;
-                        boolean bIsTruthy = neighborMaterial != 0;
-
-                        if (aIsTruthy == bIsTruthy) {
-                            mask[maskIndex] = 0;
-                        } else if (aIsTruthy) {
-                            mask[maskIndex] = currentMaterial;
-                        } else {
-                            mask[maskIndex] = -neighborMaterial;
+        
+        final boolean RENDERTYPE_SOLID = chiselRenderType.layer.equals(RenderType.solid());
+        final int BITS_PER_BLOCK_SIDE = StateEntrySize.current().getBitsPerBlockSide();
+        final int BITS_PER_BLOCK_SIDE_P = BITS_PER_BLOCK_SIDE + 2;
+        final int BITS_PER_BLOCK_SIDE_P2 = BITS_PER_BLOCK_SIDE_P * BITS_PER_BLOCK_SIDE_P;
```

Comment:
Please use the code formatting that aligns with the rest.
In this particular case UPPER_CASE_SNAKE is ment for static final fields.

/me needs to add a formatting file, and gradle step.